_posts/2020-10-21-explicit-term-inference-in-scala-3.md

```diff
@@ -0,0 +1,430 @@
+---
+layout: blog-detail
+post-type: blog
+by: Meriam Lachkar, Vincenzo Bazzucchi, Scala Center
+title: Explicit term inference with Scala 3
+---
+
+# Explicit term inference with Scala 3
+
+One of the most striking changes for developers adopting Scala 3 is the introduction
+of new syntaxes to replace the implicit mechanism used in previous Scala versions.
+
+The motivation behind the new syntax is that the same `implicit` keyword was
+used for different purposes and patterns and thus it became a way to express *how* to implement
+patterns. This means that when encountering this dreaded incantation, users need to decipher what
```

Comment:
"dreaded" seems a bit dramatic, and a bit harsh on Scala 2 IMO. perhaps "shape-shifting" or "ambiguous" or something?

Scala 2 implicits weren't bad, it's just that we learned from the experience and now we know how to do even better.